test/integration/custom-fetcher.test.ts

```diff
@@ -0,0 +1,70 @@
+import { expect, sinon } from "../test-helper"
+import fetchMock = require("fetch-mock")
+import { Model, SpraypaintBase } from "../../src"
+
+describe("Custom fetcher", () => {
+  let spy1: sinon.SinonSpy
+  let spy2: sinon.SinonSpy
+
+  let ApplicationRecord: typeof SpraypaintBase
+  let Author: typeof SpraypaintBase
+  let Person: typeof SpraypaintBase
+
+  beforeEach(() => {
+    const fetcher = (a: any, b: any) =>
+      Promise.resolve(
+        new Response(
+          JSON.stringify({
+            data: {
+              id: "1",
+              type: "people",
+              attributes: {
+                firstName: "John"
+              }
+            }
+          })
+        )
+      )
+
+    spy1 = sinon.spy(fetcher)
+    spy2 = sinon.spy(fetcher)
+
+    @Model({
+      baseUrl: "http://example.com",
+      apiNamespace: "/api/v1"
+    })
+    class Base extends SpraypaintBase {}
```

Comment:
Mind adding a spec which ensures that setting a single fetcher in the base class will allow both subclasses to use it?